File: src/main/java/com/dh/ondot/schedule/api/response/ScheduleDetailResponse.java
package com.dh.ondot.schedule.api.response;

import com.dh.ondot.core.util.DateTimeUtils;
import com.dh.ondot.schedule.api.request.PlaceDto;
import com.dh.ondot.schedule.domain.Schedule;

import java.time.LocalDateTime;
import java.util.List;

public record ScheduleDetailResponse(
        Long scheduleId,
        String title,
        boolean isRepeat,
        List<Integer> repeatDays,
        LocalDateTime appointmentAt,
        AlarmDto preparationAlarm,
        AlarmDto departureAlarm,
        PlaceDto departurePlace,
        PlaceDto arrivalPlace
) {
    public static ScheduleDetailResponse from(Schedule s) {
        return new ScheduleDetailResponse(
                s.getId(), s.getTitle(), s.getIsRepeat(),
                s.getRepeatDays() == null ? List.of() : List.copyOf(s.getRepeatDays()),
                DateTimeUtils.toSeoulDateTime(s.getAppointmentAt()),
                AlarmDto.of(s.getPreparationAlarm()),
                AlarmDto.of(s.getDepartureAlarm()),
                PlaceDto.from(s.getDeparturePlace()),
                PlaceDto.from(s.getArrivalPlace())
        );
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/presentation/response/ScheduleDetailResponse.kt
package com.dh.ondot.schedule.presentation.response

import com.dh.ondot.core.util.TimeUtils
import com.dh.ondot.schedule.domain.enums.TransportType
import com.dh.ondot.schedule.presentation.request.PlaceDto
import com.dh.ondot.schedule.domain.Schedule
import java.time.LocalDateTime

data class ScheduleDetailResponse(
    val scheduleId: Long?,
    val title: String?,
    val isRepeat: Boolean,
    val repeatDays: List<Int>,
    val appointmentAt: LocalDateTime?,
    val preparationAlarm: AlarmDto,
    val departureAlarm: AlarmDto,
    val departurePlace: PlaceDto,
    val arrivalPlace: PlaceDto,
    val transportType: TransportType,
) {
    companion object {
        @JvmStatic
        fun from(s: Schedule): ScheduleDetailResponse {
            return ScheduleDetailResponse(
                scheduleId = s.id,
                title = s.title,
                isRepeat = s.isRepeat,
                repeatDays = s.repeatDays?.toList() ?: emptyList(),
                appointmentAt = TimeUtils.toSeoulDateTime(s.appointmentAt),
                preparationAlarm = AlarmDto.of(s.preparationAlarm!!),
                departureAlarm = AlarmDto.of(s.departureAlarm!!),
                departurePlace = PlaceDto.from(s.departurePlace!!),
                arrivalPlace = PlaceDto.from(s.arrivalPlace!!),
                transportType = s.transportType,
            )
        }
    }
}
